### ISSUE TITLE
[BUG] Incorrect shape for `s3_client.put_object`

### ISSUE THREAD

**Issue by geekingfrog** [vemel/mypy_boto3_builder#28]
**Describe the bug**
This is a follow up on #19 It fixed the output shape, but now the input shape doesn't accept `bytes`, only `IO[bytes]`.

I had a look at the botocore json definitions and there's no distinction there between `PutObjectRequest` and `GetObjectOutput` unfortunately. I believe a fix would require to check against [`shape.name`](https://github.com/vemel/mypy_boto3_builder/blob/master/mypy_boto3_builder/parsers/shape_parser.py#L168) and pass it down to `_parse_shape`.
This way, we could special case some operations.

That feels hackish and brittle though, so I wonder if you have any other idea?

**To Reproduce**
1. Install 'boto3-stubs[...]'
2.  Run mypy on the following code sample

```python
import boto3
from mypy_boto3 import s3

client: s3.S3Client = boto3.client("s3")
client.put_object(Bucket="my-bucket", Body=b"Hellow", Key="my-key")

```
3. Actual mypy output

```
error: Argument "Body" to "put_object" of "S3Client" has incompatible type "bytes"; expected "IO[bytes]
```

**Expected mypy-output**
All good. `Body` should have the type `Union[bytes, IO[bytes]]`

**Additional context**
`boto3-stubs` version `1.13.24.2`

**Comment by vemel** [comment]
Thank you for digging so deep into my code :) Yes, i believe only hardcode can solve this issue. Well, clear distinction is that before we talked about a return type annotation, and this is an argument type annotation. Maybe I can apply `blob` fix only to return types and leave argument types as unions.

**Comment by geekingfrog** [comment]
Yes, that's probably a good idea. I think you can restrict that to only the s3 method which return a `StreamingBody`. There may be a complication around the `resource.get` method.
```python
s3 = boto3.resource("s3")
bucket = s3.Bucket("bucket")
bucket.get("object-key") # returns a StreamingBody
```
I'm not sure how to handle that case.

**Comment by vemel** [comment]
Fixed in the latest release `1.13.26.1`. Look good to me, but let me know if you still find any issues.

I believe the code should be

```python
s3 = boto3.resource("s3")
bucket = s3.Bucket("bucket")
bucket.Object("object-key").get()["Body"] # returns IO[bytes]
```

**Comment by geekingfrog** [comment]
Brilliant, thanks for the quick update. It seems to be all good, at least for my basic usage of S3.
